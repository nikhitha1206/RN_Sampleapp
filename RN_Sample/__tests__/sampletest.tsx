// __tests__/Intro-test.js
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import React from 'react';
import { Button } from "react-native";
import renderer from 'react-test-renderer';
import App from '../App';
import { configure,shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
describe("add new goal button test",()=>{
  const onPress = jest.fn();
    test("button test", () => {
      const { getByText } = render(<App />);
      const btn = getByText("Add new Goal");
      expect(btn).not.toBeNull();
    });
    test("number of buttons test",()=>{
      const { getByText }=render(<App />);
      const btn=getByText("Add new Goal");
      expect(btn.findAllByType).toHaveLength(1)
    });
    test("button on press",()=>{
      const { getByText }=render(<Button title="Add new Goal" onPress={onPress} />)
    });
})