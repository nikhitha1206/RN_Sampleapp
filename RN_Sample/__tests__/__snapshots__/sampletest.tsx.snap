// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 50,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add new Goal
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Course Goal"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "margin": 10,
            "padding": 10,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "60%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "red",
                    },
                  ]
                }
              >
                CANCEL
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                ADD
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
